module-bluetooth/Bluetooth/interface/profiles/HFP/HFP.cpp

```diff
@@ -483,20 +472,28 @@ namespace bluetooth
     }
     void HFP::HFPImpl::startRinging() const noexcept
     {
+        LOG_DEBUG("Starting incoming call");
+        isIncomingCall = true;
         hfp_ag_incoming_call();
     }
     void HFP::HFPImpl::stopRinging() const noexcept
-    {}
+    {
+        LOG_DEBUG("Stop ringing called!");
+    }
     auto HFP::HFPImpl::callAnswered() const noexcept -> Error::Code
     {
-        LOG_DEBUG("Call answered!");
-        hfp_ag_answer_incoming_call();
-        establishAudioConnection();
-        if (!isAudioRouted) {
-            LOG_DEBUG("Routing unrouted audio after answering call");
+        if (isIncomingCall) {
+            LOG_DEBUG("Answering incoming call");
+            hfp_ag_answer_incoming_call();
+        }
+        else {
+            LOG_DEBUG("Establishing outgoing call");
             audioInterface->startAudioRouting(const_cast<sys::Service *>(ownerService));
-            isAudioRouted = true;
+            hfp_ag_outgoing_call_established();
         }
+        LOG_DEBUG("Establishing HFP audio connection");
+        hfp_ag_establish_audio_connection(aclHandle);
+
         return Error::Success;
     }
```

Comment:
maybe it's worth to rename this funciton to sth like `setupCall`?

Reply:
Done!